src/exercises/01-use-persist/hook.ts

```diff
@@ -1 +1,24 @@
-export const usePersist = () => null
+import * as React from 'react'
+
+export const usePersist = <TValue>(
+  key: string,
+  value: TValue,
+): [TValue, React.Dispatch<React.SetStateAction<TValue>>] => {
+  const [state, setState] = React.useState(() => {
```

Comment:
How can type casting really help us?
For Example, in this code block

```js
interface User {
  firstName: string;
  lastName: string;
  accountBalance: number;
}

const json =
  '{"firstName": "Kevin", "lastName": "Le", "accountBalance": "100"}';
const user = JSON.parse(json) as User;

console.log(user); // { firstName: 'Kevin', lastName: 'Le', accountBalance: '100' }
console.log(typeof user.accountBalance); // string
```

while developing, we get the type of `user.accountBalance` is a number, but it is a string still.

Reply:
https://medium.com/ingeniouslysimple/typescript-casts-are-not-casts-4c6c954749c4
Since typescript type casts are just "type assertions" and are not actual casts/conversions. So, using this can harm us. And you also suggest to not use type casts as much as possible in our app too! :sweat_smile: 

Is there any way which actually casts the output to required type? :thinking:

Reply:
we need to use a run-time validation like [zod](http://zod.dev) to do runtime assertions to * make sure * the untrusted data is not corrupted, and the data adheres to our expected shape. A working example [ts/playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwF4F84DMoRHA5EiAE1wChRJY4AqOAQwGc4AlAU1oGN5Ns8o3PSJFgA8K8dhAB29eAFd6LAAoso9YDLgBeOAB4AKgBoAfAAoScOAGsWATwBccGVGCSA5gfNwXwGMFoAbADUA2RYHQ096dgALFhBaByQAOgAtIj0bMBZ9IxIASgcAbUNmfhgkgBF1MFoYGJ1WDnKAZRYYZphalgBBTmApHKMAXS0jRE8JaXhCgDcQlgNHNuD-UJHtRs4k+RYOrpMTPNHxiwtJjR84rTgAdxdCCBuk-wh2AI7oWlcWJO+YAEkYHETNYbHkSJ4LMB0HATJcQEcEJDTjAoDYTqdMXwYLIoJJHDE4rQkjVVCwTAApZoAeQAciTaGS4UCEeDMRY0G86tEMey4NjcfjvL4AitQsiOciUMiBXivJIfH4gvNPChwZ5NuUdgBRdDoFicA5HTRjJGYu6SB5PF5vfwfKBfH4KAEskG2RZUulJJwuVzQmwmOarFh5NlwFCLQqgxZBtbqiyy-GzeaLZ1ilhDEjSkjneAAWQgOwAGgBNa7JCAAIwAVgaYCYzadhIkkpJZCBKypDh5MfY4Mk2x2u+C1SQYJkWHAC8Wy9pki59VAdOOshAYdOFKXcjmpBoAMLYMDXQ7HRtnXfTGRdVNtPZA9ZwHbKVTqGA6DcsLcmXAgQsKAC0wg2LgiwIM2cAAAyLH2EERlOf6fiW8anImcBtv4-hZiQQA)